src/Serialization/Rings.jl

```diff
@@ -0,0 +1,112 @@
+################################################################################
+#  non simpleton base rings
+
+function load_internal(s::DeserializerState, t::Type{Nemo.NmodRing}, dict::Dict)
+    modulus = load_type_dispatch(s, UInt64, dict[:modulus])
+    
+    return t(modulus)
+end
+
+function save_internal(s::SerializerState, R::Nemo.NmodRing)
+    return Dict(
+        :modulus => save_type_dispatch(s, modulus(R))
+    )
+end
+
+################################################################################
+#  Polynomial Rings
+function save_internal(s::SerializerState, R::Union{MPolyRing, PolyRing})
+    return Dict(
+        :symbols => save_type_dispatch(s, symbols(R)),
+        :base_ring => save_type_dispatch(s, base_ring(R)),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{MPolyRing, PolyRing}},
+                       dict::Dict)
+    base_ring = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
+    symbols = load_type_dispatch(s, Vector{Symbol}, dict[:symbols]) 
+
+    if length(symbols) == 1
```

Comment:
I added the type check instead of number of symbols check, but I dont think the assert is necessary since it will throw an error if number of symbols is greater than 1 anyway